### ISSUE TITLE
Failed: FlowPrefixAndTailSpec

### ISSUE THREAD

**Issue by johanandren** [akka/akka#29965]
https://jenkins.akka.io:8498/job/pr-validator-per-commit-jenkins/21325/testReport/junit/akka.stream.scaladsl/FlowPrefixAndTailSpec/PrefixAndTail_must_throw_if_tail_is_attempted_to_be_materialized_twice/

```
org.scalatest.exceptions.TestFailedException: false was not equal to true
	at org.scalatest.matchers.MatchersHelper$.indicateFailure(MatchersHelper.scala:339)
	at org.scalatest.matchers.should.Matchers$AnyShouldWrapper.shouldBe(Matchers.scala:6982)
	at akka.stream.scaladsl.FlowPrefixAndTailSpec.$anonfun$new$15(FlowPrefixAndTailSpec.scala:106)
	at akka.stream.testkit.scaladsl.StreamTestKit$.assertAllStagesStopped(StreamTestKit.scala:32)
	at akka.stream.scaladsl.FlowPrefixAndTailSpec.$anonfun$new$14(FlowPrefixAndTailSpec.scala:93)
```

**Comment by johanandren** [comment]
https://jenkins.akka.io:8498/job/pr-validator-per-commit-jenkins/21332/testReport/junit/akka.stream.scaladsl/FlowPrefixAndTailSpec/PrefixAndTail_must_throw_if_tail_is_attempted_to_be_materialized_twice/

**Comment by johanandren** [comment]
Interestingly enough, the exception is thrown and has the right message, however, the assertion on the stack trace contents fails.

**Comment by andreaTP** [comment]
Forgot to mention that I hacked it around in #29956 :

https://github.com/akka/akka/pull/29956/files#diff-540d482074d05ca7838a79f33b41f5a20f94385461e161af8d6c0e55f4503c45R101-R102

the error was reproducible in that branch.

**Comment by johanandren** [comment]
Wasn't that strange when I dug into it, there's a PR with a fix in #29976
